tests/Functional/Cases/Rendering/DataAccessorTest.php

```diff
@@ -0,0 +1,207 @@
+            [
+                ['value' => 'value'],
+                ['value' => 'value'],
+            ],
+            [
+                new \ArrayObject(['value' => 'value']),
+                ['value' => 'value'],
+            ],
+            // accessing using properties directly
+            [
+                $this->createObjectWithProperties(),
+                [
+                    'privateValue' => null,
+                    'protectedValue' => null,
+                    'publicValue' => 'publicValue',
+                ],
+                'Cannot access private property class@anonymous::$privateValue', // @todo Probably regression?!
+            ],
+            [
+                $this->createObjectWithProperties(),
+                [
+                    'publicValue' => 'publicValue',
+                ],
+            ],
+            // accessing using getters
+            [
+                $this->createObjectWithGetter(),
+                [
+                    'privateValue' => 'privateValue@getPrivateValue()',
+                    'protectedValue' => 'protectedValue@getProtectedValue()',
+                    'publicValue' => 'publicValue@getPublicValue()', // @todo Probably regression?!
+                ],
+            ],
+            // accessing using magic method __call()
+            [
+                $this->createObjectWithMagicCall(),
+                [
+                    'privateValue' => null,
+                    'protectedValue' => null,
+                    'publicValue' => 'publicValue@__call("getPublicValue")',
+                ],
+                'Cannot access private property class@anonymous::$privateValue' // @todo Probably regression?!
+            ],
+            [
+                $this->createObjectWithMagicCall(),
+                [
+                    // 'privateValue' => 'privateValue@__call(getPrivateValue)',
+                    // 'protectedValue' => 'protectedValue@__call(getProtectedValue)',
+                    // 'publicValue' => 'publicValue@__call(getPublicValue)',
+                    'publicValue' => 'publicValue', // @todo Probably regression?!
+                ],
+            ],
+            // accessing using magic method __get()
+            [
+                $this->createObjectWithMagicGet(),
+                [
+                    'privateValue' => 'privateValue@__get(privateValue)',
+                    'protectedValue' => 'protectedValue@__get(protectedValue)',
+                    'publicValue' => 'publicValue', // @todo Probably regression?!
+                ],
+            ],
+            // accessing using everything
+            [
+                $this->createObjectWithEverything(),
+                [
+                    'privateValue' => 'privateValue@getPrivateValue()',
+                    'protectedValue' => 'protectedValue@getProtectedValue()',
+                    'publicValue' => 'publicValue@getPublicValue()', // @todo Probably regression?!
```

Comment:
Getter takes priority and did before, also for public properties - so this probably is not a regression.